io/zenoh-transport/src/unicast/universal/rx.rs

```diff
@@ -97,19 +97,19 @@ impl TransportUnicastUniversal {
             Reliability::BestEffort => zlock!(c.best_effort),
         };
 
-        self.verify_sn(sn, &mut guard)?;
-
-        let callback = zread!(self.callback).clone();
-        if let Some(callback) = callback.as_ref() {
-            for msg in payload.drain(..) {
-                self.trigger_callback(callback.as_ref(), msg)?;
+        if self.verify_sn(sn, &mut guard)? {
+            let callback = zread!(self.callback).clone();
+            if let Some(callback) = callback.as_ref() {
+                for msg in payload.drain(..) {
+                    self.trigger_callback(callback.as_ref(), msg)?;
+                }
+            } else {
+                tracing::debug!(
+                    "Transport: {}. No callback available, dropping messages: {:?}",
+                    self.config.zid,
+                    payload
+                );
             }
-        } else {
```

Comment:
I would keep the `else` statement but at `trace` level.

Reply:
This is already the case within the `verify_sn` function. When it returns `Ok(false)` it also logs to `trace` that a message was dropped

Reply:
Ok, see my most recent comment on nesting.